app/src/main/java/ch/epfl/sweng/favors/database/User.java

```diff
@@ -61,6 +62,15 @@ public User(FirebaseAuth instance){
         }
     }
 
+    public User(FirebaseAuth instance, FirebaseFirestore db){
```

Comment:
Is this to be used only for testing? If so we should have a singleton class that has a Boolean debugState that defaults to false. A test can set this singleton to true and then this would allow the call of these custom methods that are present for testing. Otherwise we are opening ourselves up to issues that people could misuse the classes by using constructors that are only present for testing purposes. 
If we have a TestSingleton.testMode we can easily as the first instruction in the class do

If(TestSingleton.testMode) throw new Exception.....

This would avoid the misuse of the test parts of the classes

Reply:
```Java
final class TestMode{
private static boolean testMode = false;
public static void setTestMode(boolean b){
testMode = b;
}

public static void getTest(){
return testMode;
}
```